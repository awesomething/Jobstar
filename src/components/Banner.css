@media (max-width: 800px) {
    .banner {
      max-width: 375px;
      flex-wrap: wrap;
    }
  }
  
  .banner {
    background: url("https://i.pinimg.com/564x/c0/03/15/c00315da0588f81c3d65e23a692346c7.jpg") center center;
    height: 20vh;
    position: relative;
  }
  
  .banner__info {
      background-color: black;
      color: white;
      padding-top: 5vh;
      padding-left: 50px;
      padding-right: 50px;
      padding-bottom: 50px;
      margin-left: 6%;
      width: fit-content;
      height: fit-content;
      /* border-radius: 40%; */
   
    overflow: hidden;
  }
  
  /*.banner__search {
      display: flex;
      flex-direction: column;
  }
  
  .banner__searchButton {
    background-color: white !important;
    font-weight: 900 !important;
    text-transform: inherit !important;
    color: #0011ff !important;
  }*/
  
  .banner__info > button {
      background-color: black;
      align-items:unset;
      text-align: center;
      color: #eeff00;
      justify-content: center;
      flex-direction: column;
      text-transform: inherit;
      /* margin-top: 20px; */
      font-weight: 600;
      border-radius: 10%;
    position: relative;
    overflow: hidden;
  }
  
  .banner__info > button:hover {
      background-color: rgb(8, 8, 8);
      color: #ff7779;
    }
    
  
  .banner__info > h5 {
      margin-top: 10px;
  }