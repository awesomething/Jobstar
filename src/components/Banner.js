import React, { useState } from 'react'
import './Banner.css'
import { Button } from "@material-ui/core";
import Search from './Search';
//import JobListMain from './JobListMain/JobListMain'
import { useHistory } from "react-router-dom";

function Banner() {
    const history = useHistory();
    const [showSearch, setShowSearch] = useState(false);

    return (
        <div className='banner'>
            <div className='banner__search'>
                {showSearch && <Search/>}

                <Button onClick={() => setShowSearch(!showSearch)} className='banner__searchButton' variant='outlined'>
                    {showSearch ? "Hide" : "Search"}
                </Button>
            </div>
            <div className='banner__info'>
            
              <h3>Are You Looking To Hire Developers On Contract?</h3>
              <h5>Post a contract job and hire top candidates.</h5>
              
            
                <Button onClick={() => history.push('/search')} variant='outlined'>Post Jobs</Button>
            </div>
        </div>
    )
}

export default Banner