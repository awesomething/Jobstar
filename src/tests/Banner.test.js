import React from 'react';
import { shallow } from 'enzyme'
//import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import Banner from '../components/Banner'

Enzyme.configure({ adapter: new Adapter() });

describe(`Banner component`, () => {
  const props = {
    id: 'a',
    name: 'test-class-name',
    
  }

  it('renders a .Banner by default', () => {
    const wrapper = shallow(<Banner />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the Banner given props', () => {
    const wrapper = shallow(<Banner {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
