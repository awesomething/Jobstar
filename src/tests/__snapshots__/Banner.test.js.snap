// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component renders a .Banner by default 1`] = `
<div
  className="banner"
>
  <div
    className="banner__search"
  />
  <div
    className="banner__info"
  >
    <h3>
      Are You Looking To Hire Developers On Contract?
    </h3>
    <h5>
      Post a contract job and hire top candidates.
    </h5>
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
      variant="outlined"
    >
      Post Jobs
    </WithStyles(ForwardRef(Button))>
  </div>
</div>
`;

exports[`Banner component renders the Banner given props 1`] = `
<div
  className="banner"
>
  <div
    className="banner__search"
  />
  <div
    className="banner__info"
  >
    <h3>
      Are You Looking To Hire Developers On Contract?
    </h3>
    <h5>
      Post a contract job and hire top candidates.
    </h5>
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
      variant="outlined"
    >
      Post Jobs
    </WithStyles(ForwardRef(Button))>
  </div>
</div>
`;
